### PR TITLE
[Feature] Implement `Streaming Mask` and add `Block Sparse Interface`.

### PR DESCRIPTION
**Pull request by KuangjuX**
This PR primarily implemented the following features:

- Streaming Attention Mask: Implemented `apply_streaming_mask` based on the `AttentionMask` class to support masking operations in Streaming Attention.

- Block Sparse Attention Interface: Added interfaces for block sparse attention and streaming attention (paged attention support will follow).

- Testing and Validation: Built a minimal CuteDSL kernel and verified the correctness of `apply_streaming_mask`; implemented a PyTorch version of reference streaming attention.

In the next PR, I will support Streaming Mask for FlashAttention and verify its correctness.

### CODE REVIEW

File: python/sglang/test/attention/test_streaming_attention.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+                                形状为 (total_tokens, num_heads, head_dim)。
+        cu_seqlens_q, cu_seqlens_k (torch.Tensor): 累积序列长度。
+                                                  例如, 对于长度为 [L1, L2] 的批处理, cu_seqlens 为 [0, L1, L1+L2]。
+                                                  用于从 q, k, v 中切分出每个序列。
+        max_seqlen_q, max_seqlen_k (int): 批处理中最大的查询/键序列长度。
+        head_mask_type (torch.Tensor): 一个形状为 (num_heads,) 的张量，决定每个头的注意力类型。
+                                       0: Dense Attention
+                                       <0: Streaming Attention
+                                       >0: Block Sparse Attention (此处未实现)
+        streaming_info (torch.Tensor): 一个形状为 (num_heads * 2,) 的扁平化张量，
```
Comment (yukavio) [inline python/sglang/test/attention/test_streaming_attention.py]: 这里不用是一个 tensor。就是两个数字就好。sink_size / local_size，每个 stream head 的这两个参数应该是一样的

File: python/sglang/srt/sparse_attention/kernels/attention/block_sparse_attention_interface.py [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+            is_causal,
+            exact_streaming,
+            return_softmax=False,
+            window_size_left=window_size_left,
+            window_size_right=window_size_right
+        )
+        
+        # Save all tensors and parameters needed for backward pass
+        # These will be retrieved in the backward() method via ctx.saved_tensors
+        ctx.save_for_backward(q, k, v,
```
Comment (yukavio) [inline python/sglang/srt/sparse_attention/kernels/attention/block_sparse_attention_interface.py]: 这里不用了，我们不用支持反向

File: python/sglang/test/attention/test_streaming_attention.py (hunk 1 above)
Comment (KuangjuX) [reply python/sglang/test/attention/test_streaming_attention.py]: fixed.

File: python/sglang/test/attention/test_streaming_attention.py [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+
+        output[q_start:q_end] = out_batch
+
+        if return_attn_probs:
+            attn_weight_lists.append(attn)
+
+    if return_attn_probs:
+        return output, attn_weight_lists
+    else:
+        return output, None
```
Comment (yukavio) [inline python/sglang/test/attention/test_streaming_attention.py]: 后续需添加 attention op级别的测试。
需要支持的 feature：
1. paged attention
2. chunked_prefill

File: python/sglang/srt/sparse_attention/kernels/attention/mask.py [hunk 4]
```diff
@@ -286,15 +272,11 @@ def apply_mask_sm100(
                 if cutlass.const_expr(self.window_size_right is not None):
                     col_limit_right = row_idx + local_row_offset_right
                     if cutlass.const_expr(mask_seqlen):
-                        col_limit_right = cutlass.min(
-                            col_limit_right, seqlenk_col_limit
-                        )
+                        col_limit_right = cutlass.min(col_limit_right, seqlenk_col_limit)
```
Comment (yukavio) [inline python/sglang/srt/sparse_attention/kernels/attention/mask.py:239]: 是因为我们 pre-commit 的版本没有对齐吗？没有修改的地方不要修改 code style

Comment (KuangjuX) [reply python/sglang/srt/sparse_attention/kernels/attention/mask.py:239]: Fixed.

File: python/sglang/srt/sparse_attention/kernels/attention/block_sparse_attention_interface.py (hunk 2 above)
Comment (KuangjuX) [reply python/sglang/srt/sparse_attention/kernels/attention/block_sparse_attention_interface.py]: Deleted.

File: python/sglang/test/attention/test_streaming_attention.py (hunk 3 above)
Comment (KuangjuX) [reply python/sglang/test/attention/test_streaming_attention.py]: In my next PR, I will implement `FlashAttention` using Streaming Mask. I will then test its application on **chunked prefill** and **paged attention**.

**Comment by yukavio** [comment]
LGTM
